workers/src/bin/dvn.rs

```diff
@@ -0,0 +1,118 @@
+//! Main offchain workflow for Nuff's DVN.
+
+use alloy::primitives::{Address, U256};
+use eyre::Result;
+use futures::stream::StreamExt;
+use tracing::{debug, error, info};
+use tracing_subscriber::EnvFilter;
+use workers::{
+    abi::{L0V2EndpointAbi::PacketSent, SendLibraryAbi::DVNFeePaid},
+    chain::{
+        connections::{build_subscriptions, get_abi_from_path, get_http_provider},
+        contracts::{create_contract_instance, query_already_verified, query_confirmations, verify},
+    },
+    data::dvn::Dvn,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Initialize tracing
+    tracing_subscriber::fmt()
+        .without_time()
+        .with_target(false)
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    // Initialize the DVN worker.
+    let mut dvn_data = Dvn::new_from_env()?;
+
+    // Create the WS subscriptions for listening to the events.
+    let (_provider, mut endpoint_stream, mut sendlib_stream) = build_subscriptions(&dvn_data.config).await?;
+
+    // Create an HTTP provider to call contract functions.
+    let http_provider = get_http_provider(&dvn_data.config)?;
+
+    // Get the relevant contract ABI, and create contract.
+    let receivelib_abi = get_abi_from_path("./abi/ArbitrumReceiveLibUln302.json")?;
+    let contract_address = dvn_data.config.receivelib_uln302_addr.parse::<Address>()?;
+    let receivelib_contract = create_contract_instance(contract_address, http_provider, receivelib_abi)?;
```

Comment:
shhoulkdnt this be in the same vain as ./abi?

Reply:
yeah it looks like we do two ways of loading the abi here

Reply:
```
sol!(
    #[allow(missing_docs)]
    #[sol(abi, rpc)]
    #[derive(Debug, Serialize, Deserialize)]
    L0V2EndpointAbi,
    "abi/L0V2Endpoint.json"
);
```
Allow just does the stuff you've written yourself in `get_abi_from_path`, is this just GPT code?